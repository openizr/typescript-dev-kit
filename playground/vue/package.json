{
  "license": "MIT",
  "devDependencies": {
    "@vue/test-utils": "^1.2.2",
    "typescript-dev-kit": "file:../../../../typescript-dev-kit/dist"
  },
  "dependencies": {
    "basx": "^1.3.7",
    "diox": "^4.0.5",
    "vue": "^2.6.10"
  },
  "eslintConfig": {
    "extends": [
      "./node_modules/typescript-dev-kit/main.js"
    ]
  },
  "tsDevKitConfig": {
    "target": "web",
    "devServer": {
      "ip": "0.0.0.0",
      "port": 5001
    },
    "html": {
      "template": "./html/index.html",
      "entries": [
        "typescript"
      ]
    },
    "entry": {
      "typescript": "./scripts/typescript.ts",
      "javascript": "./scripts/javascript.js"
    },
    "splitChunks": true,
    "srcPath": "src",
    "distPath": "public",
    "banner": "/*!\n * Copyright (c) Matthieu Jabbour.\n * All rights reserved.\n */",
    "env": {
      "development": {
        "NODE_ENV": "development"
      },
      "production": {
        "NODE_ENV": "production"
      }
    }
  },
  "scripts": {
    "init": "node_modules/typescript-dev-kit/scripts/init.sh",
    "dev": "cd node_modules/typescript-dev-kit/ && node scripts/dev",
    "build": "cd node_modules/typescript-dev-kit/ && node scripts/build && cd ../../ && sed -i -e \"s%/\\*WEBPACK_CSS\\*/%WEBPACK_CSS\\n%g\" public/index.html  && sed -i -e \"/WEBPACK_CSS/r $(ls public/assets/styles/common.*.css)\" public/index.html && sed -i -e \"s/WEBPACK_CSS//g\" public/index.html",
    "test": "cd node_modules/typescript-dev-kit/ && node scripts/test",
    "doc": "node_modules/.bin/typedoc --out ./docs/ --exclude \"**/__+(tests|mocks)__/**\" src/scripts/javascript.js",
    "postinstall": "rm -f node_modules/.eslintcache"
  }
}