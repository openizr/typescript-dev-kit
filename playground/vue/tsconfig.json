{
  "extends": "./node_modules/typescript-dev-kit/tsconfig.json",
  "compilerOptions": {
    "baseUrl": "src",
  },
  "exclude": [
    "../../coverage",
    "../../doc"
  ]
}