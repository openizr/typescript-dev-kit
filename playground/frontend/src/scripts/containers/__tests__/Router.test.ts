/**
 * @jest-environment jsdom
 */

import store from 'scripts/store/index';
import { render, waitFor } from '@testing-library/vue';
import Router from 'scripts/containers/Router.vue';

// Useful mocks allowing us to easily test Vue lazy components.
jest.mock('scripts/store/routes', () => ({
  // eslint-disable-next-line global-require, @typescript-eslint/no-var-requires
  '/': require('scripts/pages/Home.vue').default,
}));

describe('vue/Router', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - loading page', () => {
    const { container } = render(Router, { props: { locale: { LABEL_TEST: 'Test' } } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - page found', async () => {
    store.mutate('router', 'NAVIGATE', '/');
    const { container } = render(Router, { props: { locale: { LABEL_TEST: 'Test' } } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - page not found', async () => {
    store.mutate('router', 'NAVIGATE', '/404');
    await waitFor(() => new Promise((resolve) => {
      setTimeout(resolve, 50);
    }));
    const { container } = render(Router, { props: { locale: { LABEL_TEST: 'Test' } } });
    expect(container.firstChild).toMatchSnapshot();
  });
});
