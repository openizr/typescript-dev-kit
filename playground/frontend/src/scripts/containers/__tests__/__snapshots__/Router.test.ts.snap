// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue/Router renders correctly - loading page 1`] = `
<p>
  <img
    alt="test"
    src="file"
  />
  <img
    alt="test"
    src="file"
  />
   Test
</p>
`;

exports[`vue/Router renders correctly - page found 1`] = `
<p>
  <img
    alt="test"
    src="file"
  />
  <img
    alt="test"
    src="file"
  />
   Test
</p>
`;

exports[`vue/Router renders correctly - page not found 1`] = `<!--v-if-->`;
